src/main/java/com/smartbear/postman/collection/PostmanCollectionFactory.java

```diff
@@ -1,48 +1,41 @@
 import net.sf.json.JSONObject;
 import org.apache.commons.lang.StringUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import static com.smartbear.postman.exceptions.PostmanCollectionUnsupportedVersionException.UNSUPPORTED_VERSION_MESSAGE;
 
 public class PostmanCollectionFactory {
     public static final String INFO = "info";
     public static final String COLLECTION = "collection";
+    public static final String SCHEMA = "schema";
     public static final String VERSION_2 = "2.0.0";
     public static final String VERSION_2_1 = "2.1.0";
-    private static final String V1_IS_DEPRECATED_ERROR_MESSAGE =
-            "Postman Collections in v1 format are no longer supported. " +
-            "Convert your collection to version v2 or higher and try importing again.";
-    private static final Logger logger = LoggerFactory.getLogger(PostmanCollectionFactory.class);
 
-    public static PostmanCollection getCollection(JSONObject collectionObject) {
+    public static PostmanCollection getCollection(JSONObject collectionObject) throws PostmanCollectionUnsupportedVersionException {
         Object collectionSharedViaAPI = collectionObject.get(COLLECTION);
         if (collectionSharedViaAPI != null) {
             collectionObject = (JSONObject) collectionObject.get(COLLECTION);
         }
 
+        PostmanCollection collection = null;
         Object infoSection = collectionObject.get(INFO);
         if (infoSection != null) {
-            return getCollectionByVersion(collectionObject, (JSONObject) infoSection);
-        } else {
-            // info section is not present for collections of version 1.0.0
-            UISupport.showErrorMessage(V1_IS_DEPRECATED_ERROR_MESSAGE);
-            return null;
+            collection = getCollectionByVersion(collectionObject, (JSONObject) infoSection);
         }
+        if (collection == null) {
+            throw new PostmanCollectionUnsupportedVersionException(UNSUPPORTED_VERSION_MESSAGE);
+        }
+        return collection;
     }
 
     private static PostmanCollection getCollectionByVersion(JSONObject collectionObject, JSONObject infoSection) {
-        Object schemaField = infoSection.get("schema");
+        Object schemaField = infoSection.get(SCHEMA);
         if (schemaField instanceof String) {
             String collectionVersion = StringUtils
                     .substringBetween((String) schemaField, "collection/v", "/collection.json");
-            if (VERSION_2.equals(collectionVersion)) {
+            if (VERSION_2.equals(collectionVersion) || VERSION_2_1.equals(collectionVersion)) {
                 return new PostmanCollectionV2(collectionObject);
-            } else if (VERSION_2_1.equals(collectionVersion)) {
-                return new PostmanCollectionV2_1(collectionObject);
-            } else {
-                logger.error("Unsupported version {} of Postman collection is provided", collectionVersion);
             }
         }
         return null;
```

Comment:
I think we should throw exception here instead of returning null. The next command after returning null from this method is a null-check and throwing exception anyway.